prov/efa/src/rdm/efa_rdm_ep_utils.c

```diff
@@ -209,12 +209,10 @@ int efa_rdm_ep_post_user_recv_buf(struct efa_rdm_ep *ep, struct efa_rdm_ope *rxe
 {
 	struct efa_rdm_pke *pkt_entry;
 	struct efa_mr *mr;
-	int err;
+	size_t rx_iov_offset;
+	int rx_iov_index, err;
 
-	/*
-	 * TODO remove/change assert expression when function logic is fixed
-	 */
-	assert(rxe->iov_count == 1);
+	assert(rxe->iov_count > 0  && rxe->iov_count <= ep->rx_iov_limit);
```

Comment:
This still doesn't make us safe if application tries to post a recv with iov count 4 (the current rx_iov_limit). I think a better fix should be reducing the rx iov limit to 2 when FI_MSG_PREFIX is used.

Reply:
Ignore this comment, I don't think we can naively require 2 iov limit for FI_MSG_PREFIX as user may not want to do zcpy recv.